hudi-utilities/src/test/java/org/apache/hudi/utilities/functional/TestHoodieDeltaStreamer.java

```diff
@@ -1159,6 +1163,72 @@ public void testCsvDFSSourceNoHeaderWithSchemaProviderAndTransformer() throws Ex
     testCsvDFSSource(false, '\t', true, Collections.singletonList(TripsWithDistanceTransformer.class.getName()));
   }
 
+  @Test
+  public void testDFSSourceDeleteFilesAfterCommit() throws Exception {
+    TypedProperties props = new TypedProperties();
+    props.setProperty("hoodie.deltastreamer.source.dfs.clean.mode", "delete");
+    testDFSSourceCleanUp(props);
+  }
+
+  @Test
+  public void testDFSSourceArchiveFilesAfterCommit() throws Exception {
+    TypedProperties props = new TypedProperties();
+    props.setProperty("hoodie.deltastreamer.source.dfs.clean.mode", "archive");
+    final String archivePath = dfsBasePath + "/archive";
+    dfs.mkdirs(new Path(archivePath));
+    props.setProperty("hoodie.deltastreamer.source.dfs.clean.archiveDir", archivePath);
+    assertEquals(0, Helpers.listAllFiles(archivePath).size());
+    testDFSSourceCleanUp(props);
+    // archive dir should contain source files
+    assertEquals(1, Helpers.listAllFiles(archivePath).size());
+  }
+
+  private void testDFSSourceCleanUp(TypedProperties props) throws Exception {
+    // since each source cleanup test will modify source, we need to set different dfs.root each test
+    final String dfsRoot = dfsBasePath + "/test_dfs_cleanup_source" + testNum;
+    dfs.mkdirs(new Path(dfsRoot));
+    prepareParquetDFSFiles(dfsRoot + "/1.parquet", PARQUET_NUM_RECORDS);
+    props.setProperty("hoodie.deltastreamer.source.dfs.root", dfsRoot);
+    prepareParquetDFSSource(false, false, props);
```

Comment:
can we just move this config within prepareParquetDFSSource() itself. so that we don't need to make changes to prepareDFSSource signature.